Dockerfile

```diff
@@ -1,24 +1,80 @@
+# from source or manually installing inside a Docker container.
+# 
+# To disregard this advice and make this Dockerfilework for your GPU,
+# select a new parent image from:
+# https://catalog.ngc.nvidia.com/orgs/nvidia/containers/pytorch/tags
+# and then update the installation steps to install the appropriate
+# versions of PyTorch:
+# https://pytorch.org/get-started/locally/
+# and PyTorch Scatter:
+# https://github.com/rusty1s/pytorch_scatter?tab=readme-ov-file#installation
+# based on your GPU version.
+#
+# We have absolutely no idea how to make the above work on AMD GPUs... ¯\_(ツ)_/¯
+# Good luck!
 
-USER root
+# Parent Image
+FROM ubuntu:latest
 
+# 'Install' Bash shell
+RUN ln -snf /bin/bash /bin/sh
+
+# Install system dependencies
+#
+# List of deps and why they are needed:
+#  - git for downloading repository
+#  - wget for downloading conda install script
+#  - libxrender1 required by RDKit
 RUN apt-get update && \
-    apt-get install -y git && \
-    rm -rf /var/lib/{apt,dpkg,cache,log}
+    apt-get install -y \
+    wget \
+    git \
+    libxrender1 && \
+    apt-get autoremove -y && \
+    apt-get clean -y
 
-USER $MAMBA_USER
+# Install conda
+RUN wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh && \
+    bash Miniconda3-latest-Linux-x86_64.sh -b -p /miniconda && \
+    rm Miniconda3-latest-Linux-x86_64.sh
+ENV PATH="$PATH:/miniconda/bin"
 
-COPY --chown=$MAMBA_USER:$MAMBA_USER environment.yml /tmp/environment.yml
+# Set Bash as the default shell for following commands
+SHELL ["/bin/bash", "-c"]
 
-RUN micromamba install -y -n base -f /tmp/environment.yml && \
-    micromamba clean --all --yes
+WORKDIR /opt/chemprop
 
-COPY --chown=$MAMBA_USER:$MAMBA_USER . /opt/chemprop
+# build the conda environment
+RUN conda create --name chemprop_env python=3.11* && \
+    conda clean --all --yes
 
-WORKDIR /opt/chemprop
+# This runs all subsequent commands inside the chemprop_env conda environment
+#
+# Analogous to just activating the environment, which we can't actually do here
+# since that requires running conda init and restarting the shell (not possible
+# in a Dockerfile build script)
+SHELL ["conda", "run", "--no-capture-output", "-n", "chemprop_env", "/bin/bash", "-c"]
 
-RUN /opt/conda/bin/python -m pip install torch-scatter -f https://data.pyg.org/whl/torch-2.2.0+${CUDA}.html && \
-    /opt/conda/bin/python -m pip install torch && \
-    /opt/conda/bin/python -m pip install --upgrade-strategy only-if-needed -e .
+# Follow the installation instructions (but with fixed versions, for stability of this image) then clear the cache
+RUN python -m pip install torch==2.2.0 --index-url https://download.pytorch.org/whl/cpu && \
+    python -m pip install torch-scatter -f https://data.pyg.org/whl/torch-2.2.0+cpu.html && \
+    python -m pip install git+https://github.com/chemprop/chemprop.git@f39a672d003dd82f1fddff8009d98a0c4f21796b && \
+    python -m pip cache purge
```

Comment:
Can we use `ARG`s here, i.e.
```suggestion
# Follow the installation instructions (but with fixed versions, for stability of this image) then clear the cache
RUN python -m pip install torch==${TORCH} --index-url https://download.pytorch.org/whl/${CUDA} && \
    python -m pip install torch-scatter -f https://data.pyg.org/whl/torch-${TORCH}+${CUDA}.html && \
    python -m pip install git+https://github.com/chemprop/chemprop.git@f39a672d003dd82f1fddff8009d98a0c4f21796b && \
    python -m pip cache purge
```

Reply:
Feel free to work this out and push it, I think actually running the image with GPU dependencies included will require setting up https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/latest/install-guide.html which I haven't done and I think only works on debian linux.

Reply:
yeah good point- I'm no expert on docker and I don't think it's worth it to spend time figuring this out (at least in the near term). I only suggest this to clearly demarcate where we (or others) will need to go looking in the future for what to edit in the file (and ensure that there's only one source of truth for version info).